daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/ValueVersion.scala

```diff
@@ -29,7 +29,23 @@ object ValueVersions
   private[value] val minGenMap = ValueVersion("7")
   private[value] val minContractIdV1 = ValueVersion("7")
 
-  def assignVersion[Cid](v0: Value[Cid]): Either[String, ValueVersion] = {
+  abstract class VersionCid[-Cid] private[lf] {
+    private[lf] def assignVersion(cid: Cid): ValueVersion
+  }
+
+  object VersionCid extends VersionCid[ContractId] {
```

Comment:
We only seem to be using this from tests and only this particular instance of the class `VersionCid`. Is it worth having this extra machinery when we don't use it? Or am I missing something?

Reply:
In the current state, canton needs one.

Reply:
I see. thanks.